### PR TITLE
Adding SnowflakeOperator howto-documentation and example DAG

### LINKED ISSUE #11921: Add how-to Guide for Snowflake operators
**Description**

A guide that describes how to use all the operators for Snowflake (https://github.com/apache/airflow/tree/master/airflow/providers/snowflake/operators) would be useful.

Other guides are available:
https://airflow.readthedocs.io/en/latest/howto/operator/index.html

Source code for those guides are at:
https://github.com/apache/airflow/tree/master/docs/howto/operator

Are you wondering how to start contributing to this project? Start by reading our [contributor guide](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst)

Best regards,
Kaxil﻿

### PR DESCRIPTION
**Pull request by davido912**
closes #11921 
tested build locally without any problems.
@kaxil 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by kaxil** [comment]
Thanks @davido912 , I have rebased your PR on latest Master, since we had an issue over there.

**Comment by davido912** [comment]
@kaxil When referring to latest master, we just mean the latest fetched and pulled changes to the master branch, correct?
Oddly enough, I fetched all changes, pulled to master and branched from there, anything in that order that was supposed to be done differently (out of curiosity)?

**Comment by kaxil** [comment]
> @kaxil When referring to latest master, we just mean the latest fetched and pulled changes to the master branch, correct?
> Oddly enough, I fetched all changes, pulled to master and branched from there, anything in that order that was supposed to be done differently (out of curiosity)?

Yea that is correct. We identified the problem after you might have created your PR, so was our issue actually :) We fixed it in https://github.com/apache/airflow/pull/11981 and merged it

File: airflow/example_dags/example_snowflake_operator.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#
```
Comment (kaxil) [inline airflow/example_dags/example_snowflake_operator.py]: We already have an example_dag: https://github.com/apache/airflow/blob/master/airflow/providers/snowflake/example_dags/example_snowflake.py

we can just use that one or just update that one.

We also store any provider-specific example_dags in that provider's folder itself

**Comment by kaxil** [comment]
Let me know if you need help

File: airflow/example_dags/example_snowflake_operator.py (hunk 1 above)
Comment (davido912) [reply airflow/example_dags/example_snowflake_operator.py]: @kaxil So I understand better, is the howto supposed to cover only the SnowflakeOperator or also all the other examples contained in the already existing example DAG?
In the issue only the SnowflakeOperator was linked, hence the confusion

Comment (kaxil) [reply airflow/example_dags/example_snowflake_operator.py]: Covering all the operators in the existing would be ideal. Although if you don't know about other operators you can just do it for SnowflakeOperator.

Comment (davido912) [reply airflow/example_dags/example_snowflake_operator.py]: Could do about all of them, would probably rework the PR tomorrow and correct where the CI complained about syntax/empty spaces. I assume we're keeping all three then in the same example DAG, I might change it a bit with the additions I had but I'll try to keep it's general structure (I didn't really understand what was the benefit of using env for all Snowflake relevant metadata in the current structure of the example DAG)

Comment (kaxil) [reply airflow/example_dags/example_snowflake_operator.py]: Yea feel free to update and change and let me know if you have any questions

**Comment by davido912** [comment]
@kaxil mind explaining what does the PyLint black test checks? 
what could be a reason for failure? can't find anything descriptive and specific there

**Comment by kaxil** [comment]
> @kaxil mind explaining what does the PyLint black test checks?
> what could be a reason for failure? can't find anything descriptive and specific there

We use Static Code checks (docs: https://github.com/apache/airflow/blob/master/STATIC_CODE_CHECKS.rst). One of them is `black`(https://black.readthedocs.io/en/stable/) which is a Python formatted.

To run `black` and all other static checks we use locally, run `pre-commit install` (https://github.com/apache/airflow/blob/master/STATIC_CODE_CHECKS.rst#id4). Pre-commit (https://pre-commit.com/) framework runs this kind of checks when you do `git commit`

**Comment by davido912** [comment]
@kaxil I thought Black as a module formats things by itself. So in this case the job fails and changes need to be done locally? 
I saw you pushed a commit changing, so let's see how that one goes

**Comment by kaxil** [comment]
> @kaxil I thought Black as a module formats things by itself. So in this case the job fails and changes need to be done locally?
> I saw you pushed a commit changing, so let's see how that one goes

It only formats automatically on git commit if you have pre-commits installed or run `pre-commit run black -a`

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
Well done :)

**Comment by potiuk** [comment]
Indeed !

**Comment by davido912** [comment]
:) !

**Pull request merged**